pandas/tests/copy_view/test_setitem.py

```diff
@@ -66,26 +67,82 @@ def test_set_column_with_index(using_copy_on_write):
 
     df["d"] = idx
 
-    assert not np.shares_memory(df["d"].values, arr)
+    assert not np.shares_memory(get_array(df, "d"), arr)
     arr[0] = 0
     tm.assert_series_equal(df["d"], Series([1, 2, 3], name="d"))
 
 
 def test_set_columns_with_dataframe(using_copy_on_write):
     # Case: setting a DataFrame as new columns copies that data
+    # (with delayed copy with CoW)
     df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
     df2 = DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
 
     df[["c", "d"]] = df2
 
     if using_copy_on_write:
-        # TODO(CoW) with CoW we can delay the copy
-        # assert np.shares_memory(df["c"].values, df2["c"].values)
-        assert not np.shares_memory(df["c"].values, df2["c"].values)
+        assert np.shares_memory(get_array(df, "c"), get_array(df2, "c"))
     else:
         # the data is copied
-        assert not np.shares_memory(df["c"].values, df2["c"].values)
+        assert not np.shares_memory(get_array(df, "c"), get_array(df2, "c"))
 
     # and modifying the set DataFrame does not modify the original DataFrame
     df2.iloc[0, 0] = 0
     tm.assert_series_equal(df["c"], Series([7, 8, 9], name="c"))
+
+
+def test_setitem_series_no_copy(using_copy_on_write):
+    # Case: setting a Series as column into a DataFrame can delay copying that data
+    df = DataFrame({"a": [1, 2, 3]})
+    rhs = Series([4, 5, 6])
+    rhs_orig = rhs.copy()
+
+    # adding a new column
+    df["b"] = rhs
+    if using_copy_on_write:
+        assert np.shares_memory(get_array(rhs), get_array(df, "b"))
+
+    df.iloc[0, 1] = 100
+    tm.assert_series_equal(rhs, rhs_orig)
+
+
+def test_setitem_series_no_copy_single_block(using_copy_on_write):
+    # Overwriting an existing column that is a single block
+    df = DataFrame({"a": [1, 2, 3], "b": [0.1, 0.2, 0.3]})
+    rhs = Series([4, 5, 6])
+    rhs_orig = rhs.copy()
+
+    df["a"] = rhs
+    if using_copy_on_write:
+        assert np.shares_memory(get_array(rhs), get_array(df, "a"))
+
+    df.iloc[0, 0] = 100
+    tm.assert_series_equal(rhs, rhs_orig)
+
+
+def test_setitem_series_no_copy_split_block(using_copy_on_write):
+    # Overwriting an existing column that is part of a larger block
+    df = DataFrame({"a": [1, 2, 3], "b": 1})
```

Comment:
I was just thinking about the other issue to no longer use copy=True as default for dict (https://github.com/pandas-dev/pandas/issues/52967), that such test cases as the above could start to no longer test what it is actually meant to test .. 
I know that here it's not a dict of Series, so this example might not change, but writing it as `DataFrame(np.array(..), columns=["a", "b"])` might be more robust?

Reply:
Or set copy true explicitly?